backend/apid/graphql/health.go

```diff
@@ -18,11 +21,16 @@ var _ schema.EtcdClusterMemberHealthFieldResolvers = (*etcdClusterMemberHealthIm
 // Implement ClusterHealthFieldResolvers
 //
 
-type clusterHealthImpl struct{}
+type clusterHealthImpl struct {
+	healthController EtcdHealthController
+}
 
 // Etcd implements response to request for 'etcd' field.
-func (r *clusterHealthImpl) Etcd(p graphql.ResolveParams) (interface{}, error) {
-	return p.Source, nil
+func (r *clusterHealthImpl) Etcd(p schema.ClusterHealthEtcdFieldResolverParams) (interface{}, error) {
+	timeout := time.Duration(p.Args.Timeout) * time.Millisecond
+	ctx := context.WithValue(p.Context, store.ContextKeyTimeout, timeout)
```

Comment:
Can we use `context.WithTimeout(p.Context, timeout)` here instead or is the store expecting a context with a value for `store.ContextKeyTimeout`?

Reply:
Admittedly I'm somewhat blindly following what we do in the health router: https://github.com/sensu/sensu-go/blob/0e648b821240a05b2d967b76b6e302da137b2a6f/backend/apid/routers/health.go#L46-L48

Looks like the timeout value is extracted and used for each request to the remote etcd nodes: https://github.com/sensu/sensu-go/blob/0e648b821240a05b2d967b76b6e302da137b2a6f/backend/store/etcd/health_store.go#L78-L86

Reply:
The health router is probably doing the wrong thing here, I'd follow Justin's advice

Reply:
I couldn't quite figure out why we wanted multiple timeouts, so I switched to WithTimeout as suggested. Bit of a _Chesterton's Fence_ scenario but at least this is easier to reason about.